src/System.Text.Json/tests/Serialization/PropertyNameTests.cs

```diff
@@ -293,6 +293,87 @@ public static void ExtensionDataDictionarySerialize_DoesNotHonor()
             Assert.False(obj.MyOverflow.ContainsKey("key1"));
             Assert.Equal(1, obj.MyOverflow["Key1"].GetInt32());
         }
+
+        private class ClassWithPropertyNamePermutations
+        {
+            public int a { get; set; }
+            public int aa { get; set; }
+            public int aaa { get; set; }
+            public int aaaa { get; set; }
+            public int aaaaa { get; set; }
+            public int aaaaaa { get; set; }
+
+            // 7 characters - caching code only keys up to 7.
+            public int aaaaaaa { get; set; }
+            public int aaaaaab { get; set; }
+
+            // 8 characters.
+            public int aaaaaaaa { get; set; }
+            public int aaaaaaab { get; set; }
+
+            // 9 characters - caching code doesn't differentiate past this.
```

Comment:
Should this comment on 8 characters, not 9 (`caching code doesn't differentiate past this`)?